azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_jenkins.go

```diff
@@ -0,0 +1,149 @@
+package serviceendpoint
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointJenkins schema and implementation for Jenkins service endpoint resource
+func ResourceServiceEndpointJenkins() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointJenkins, expandServiceEndpointJenkins)
+
+	r.Schema["url"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Required: true,
+		ValidateFunc: func(i interface{}, key string) (_ []string, errors []error) {
+			url, ok := i.(string)
+			if !ok {
+				errors = append(errors, fmt.Errorf("expected type of %q to be string", key))
+				return
+			}
+			if strings.HasSuffix(url, "/") {
+				errors = append(errors, fmt.Errorf("%q should not end with slash, got %q.", key, url))
+				return
+			}
+			return validation.IsURLWithHTTPorHTTPS(url, key)
+		},
+		Description: "Url for the Jenkins Repository",
+	}
+
+	r.Schema["accept_untrusted_certs"] = &schema.Schema{
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Allows the Jenkins clients to accept self-signed SSL server certificates without installing them into the TFS service role and/or Build Agent computers.",
+	}
+
+	patHashKeyP, patHashSchemaP := tfhelper.GenerateSecreteMemoSchema("password")
+	aup := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Description: "The Jenkins user name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"password": {
+				Description:      "The Jenkins password.",
+				Type:             schema.TypeString,
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			patHashKeyP: patHashSchemaP,
+		},
+	}
+
+	r.Schema["authentication_basic"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     aup,
+	}
```

Comment:
`unam/password` is the only auth type supported by  Jenkines, can we flatten it to the root , not wrap by `authentication_basic`

```suggestion
	r.Schema["username"] = &schema.Schema{
		Description: "The Jenkins user name.",
		Type:        schema.TypeString,
		Required:    true,
	}
	r.Schema["password"] = &schema.Schema{
		Description: "The Jenkins password.",
		Type:        schema.TypeString,
		Required:    true,
		Sensitive:   true,
	}
```

Reply:
Done